// connected-react-router-immutable/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ConnectedRouter = exports.routerMiddleware = exports.routerActions = exports.goBackOrUrl = exports.goForward = exports.goBack = exports.go = exports.replace = exports.push = exports.GO_BACK_OR_URL = exports.CALL_HISTORY_METHOD = exports.connectRouter = exports.LOCATION_CHANGE = undefined;

var _actions = require('./actions');

Object.defineProperty(exports, 'LOCATION_CHANGE', {
  enumerable: true,
  get: function get() {
    return _actions.LOCATION_CHANGE;
  }
});
Object.defineProperty(exports, 'CALL_HISTORY_METHOD', {
  enumerable: true,
  get: function get() {
    return _actions.CALL_HISTORY_METHOD;
  }
});
Object.defineProperty(exports, 'GO_BACK_OR_URL', {
  enumerable: true,
  get: function get() {
    return _actions.GO_BACK_OR_URL;
  }
});
Object.defineProperty(exports, 'push', {
  enumerable: true,
  get: function get() {
    return _actions.push;
  }
});
Object.defineProperty(exports, 'replace', {
  enumerable: true,
  get: function get() {
    return _actions.replace;
  }
});
Object.defineProperty(exports, 'go', {
  enumerable: true,
  get: function get() {
    return _actions.go;
  }
});
Object.defineProperty(exports, 'goBack', {
  enumerable: true,
  get: function get() {
    return _actions.goBack;
  }
});
Object.defineProperty(exports, 'goForward', {
  enumerable: true,
  get: function get() {
    return _actions.goForward;
  }
});
Object.defineProperty(exports, 'goBackOrUrl', {
  enumerable: true,
  get: function get() {
    return _actions.goBackOrUrl;
  }
});
Object.defineProperty(exports, 'routerActions', {
  enumerable: true,
  get: function get() {
    return _actions.routerActions;
  }
});

var _reducer = require('./reducer');

var _reducer2 = _interopRequireDefault(_reducer);

var _middleware = require('./middleware');

var _middleware2 = _interopRequireDefault(_middleware);

var _ConnectedRouter2 = require('./ConnectedRouter');

var _ConnectedRouter3 = _interopRequireDefault(_ConnectedRouter2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.connectRouter = _reducer2.default;
exports.routerMiddleware = _middleware2.default;
exports.ConnectedRouter = _ConnectedRouter3.default;
